docs/spark/tutorials/hdinsight-deployment.md

```diff
@@ -9,124 +9,180 @@ ms.custom: mvc
+
+   There may be multiple .jar files (for versions 2.3.x and 2.4.x of Spark). You need **microsoft-spark-2.3.x-0.3.0.jar** since you chose Spark 2.3.2 during cluster creation.
+
+6. Your cluster needs the input to your app. Navigate to your **mySparkApp** directory and locate **input.txt**. Upload your input file to the **user/sshuser** directory in your blob container. You will be connecting to your cluster through ssh, and this folder is where your cluster looks for its input. The input.txt file is the only file uploaded to a specific directory.
 
-You can submit your job to Azure HDInsight using `spark-submit` or Apache Livy.
+## Run the HDInsight script action
 
-### Use spark-submit
+Once your cluster is running and you've uploaded your files to Azure, you need to run the **install-worker.sh** script on the cluster. 
 
-You can use the [spark-submit](https://spark.apache.org/docs/latest/submitting-applications.html) command to submit .NET for Apache Spark jobs to Azure HDInsight.
- 
-1. `ssh` into one of the head nodes in your cluster.
+1. Navigate to your HDInsight Spark cluster in Azure portal, and then select **Script actions**.
 
-1. Run `spark-submit`:
+2. Select **+ Submit new** and provide the following values:
+
+   |Property  |Description  |
+   |---------|---------|
+   | Script type |Custom|
+   | Name | Install Worker|
+   | Bash script URI |https://<\mystorageaccount>.blob.core.windows.net/<\mycontainer>/install-worker.sh </br> To confirm this URI, right-click on install-worker.sh in Azure Storage Explorer and select Properties. |
+   | Node type(s)| Worker|
+   | Parameters | azure </br> wasbs://<\mycontainer>@<\myStorageAccount>.blob.core.windows.net/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.3.0.tar.gz </br> /usr/local/bin 
+
+3. Select **Create** to submit your script and run on your cluster.
+
+## Run your app
+
+1. Navigate to your HDInsight Spark cluster in Azure portal, and then select **SSH + Cluster login**.
+
+2. Copy the ssh login information and paste the login into a terminal. Sign in to your cluster using the password you set during cluster creation. You should see messages welcoming you to Ubuntu and Spark.
+
+3. Use the **spark-submit** command to run your app on your HDInsight cluster. Remember to replace **mycontainer** and **mystorageaccount** in the example script with the actual names of your blob container and storage account.
 
    ```bash
-   spark-submit \
+   foo@bar:~$ $SPARK_HOME/bin/spark-submit \
+   --master yarn \
+   --class org.apache.spark.deploy.DotnetRunner \
+   wasbs://mycontainer@mystorageaccount.blob.core.windows.net/   microsoft-spark-2.3.x-0.3.0.jar \
+   wasbs://mycontainer@mystorageaccount.blob.core.windows.net/   publish.zip mySparkApp
+   
```

Comment:
The code snippets after this were things I was using/experimenting with in my demos. They repeat what's shown above, so they should be deleted.